pf4j/src/main/java/org/pf4j/util/ClassUtils.java

```diff
@@ -72,6 +77,60 @@ public static List getAllAbstractClasses(Class aClass) {
     }
     */
 
+    /**
+     * Get a certain annotation of a {@link TypeElement}.
+     * See <a href="https://stackoverflow.com/a/10167558">stackoverflow.com</a> for more information.
+     *
+     * @param typeElement the type element, that contains the requested annotation
+     * @param annotationClass the class of the requested annotation
+     * @return the requested annotation or null, if no annotation of the provided class was found
+     * @throws NullPointerException if <code>typeElement</code> or <code>annotationClass</code> is null
```

Comment:
Do we need this line (throws) in javadoc ? It's uncommon for me to add a NPE in javadoc's `throws` section. It's clear for me that the parameters cannot be null. In the end we can add this clarification in @param.

Reply:
In general I think its useful to know about runtime exceptions thrown by a method without looking into the source code. If the plugin developer uses Eclipse / Netbeans / IntelliJ he should see the javadoc commment directly in his IDE. He wouldn't have to make assumptions about possible NPE's or had to look into the PF4J source code.

Of course you might also add this information into the `@param`. But I personally prefer to separate those kind of information. Theoretically there might be other runtime exceptions besides a NPE, too (e.g. `IllegalArgumentException`).

By the way: The Java API documentation also contains the runtime exception thrown by a method (e.g. [`Integer#valueOf(java.lang.String)`](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#valueOf(java.lang.String))). Therefore I assume, that this approach is quite common.

But I leave this up to you. If you prefer another way to write Javadocs, feel free to change them after you've merged this pull request or let me know about your preferred conventions.